src/ansys/simai/core/data/optimizations.py

```diff
@@ -73,6 +73,40 @@ def _run_iteration(self, parameters: Dict) -> "_OptimizationTrialRun":
         return self._client._optimization_trial_run_directory._run_iteration(self.id, parameters)
 
 
+class OptimizationResult:
+    """Result for a non-parametric optimization."""
+
+    def __init__(
+        self, optimization: "Optimization", objectives: list[Dict], geometries: list[Dict]
+    ):
+        self._optimization = optimization
+        self._objectives: list[Dict] = objectives
+        self._geometries: list[Geometry] = []
+
+        workspace_geometries = self._optimization._client.geometries.list(
+            geometries[0]["workspace_id"]
+        )
+        workspace_geometries = {geom.id: geom for geom in workspace_geometries}
+
+        geometry_ids = [geom["id"] for geom in geometries]
```

Comment:
Use an iterator, no need to build a list here
```suggestion
        geometry_ids = (geom["id"] for geom in geometries)
```